### PR TITLE
clearOptions: Don't remove already selected items

### LINKED ISSUE #1079: How to dynamically update options on type
We are using selectize to allow a user to add hashtags to a post. When the widget is initialized, I make a request to the server that populates the dropdown with trending hashtags. Then, as the user types, I would like to update the options to show tags that match their search. e.g. (this is used in Ember)

```
let s = input.selectize({
   searchField: ['name'],
   preload: true,
   options: [],
   create: function(input, callback) {
   },
   render: {
      option: function (item, escape) {
         return '<div>' + escape(item.text) + '</div>';
      }
   },
   load: (query, callback) => {
     if (!query.length) {
        return this.initializeFromTrendingTags().then((options) =>  {
           // returns an array of options [{ text: ..., value: ... }, ...]
           callback(options);
        });
     }
     // searchTags also returns an array of matched options in format [{ text: ..., value: ... }, ...]
     return this.searchTags(query).then((options) => {
        callback(options);
     });
  }
})
```

Where searchTags issues the api request and updates the options. 
However, no matter what I do I can't seem to get the options to update. Ideally I'd like the dropdown to change while open to the matched tags. I've tried doing the following:

```
selectize.clearCache('option');
selectize.clearOptions();
selectize.addOption(options);
selectize.refreshOptions(true);
```

but this just clears out the options altogether. I'm stuck. any help? 
thanks!!!

### PR DESCRIPTION
**Pull request by andrey-lizunov**
Closed #1079

### CODE REVIEW

**Comment by joallard** [comment]
Can you explain your PR? I wasn't able to understand the issue ref'd.

**Comment by andrey-lizunov** [comment]
@joallard I found that selectize accumulates options from previous searches. I upload new list of options remotely from my server and the number of selected items not limited. In this case after several searches in dropdown I have a lot of options connected with old queries which I don't need.
I tryed to call `clearOptions` before upload new list of options, but `clearOptions` calls `clear` function inside and this removes all items which was chosen before.
The problem that I want to be able to clear just list of options without removing already selected items. I talking about multiple selects.

This PR change this behavior, now the list of selected items stay unchanged, but list of options for select is cleared.

File: src/selectize.js [hunk 1]
```diff
@@ -1336,10 +1336,15 @@ $.extend(Selectize.prototype, {
...
 		self.userOptions = {};
 		self.renderCache = {};
-		self.options = self.sifter.items = {};
+		var options = self.options;
+		$.each(self.options, function(key, value) {
+			if(self.items.indexOf(key) == -1) {
+				delete options[key];
+			}
+		});
+		self.options = self.sifter.items = options;
```
Comment (brookjordan) [inline src/selectize.js:1345]: Is this not the same as:

``` javascript
$.each(self.options, function(key, value) {
  if(self.items.indexOf(key) == -1) {
    delete self.options[key];
  }
});
self.sifter.items = self.options;
```

**Comment by mtebele** [comment]
Any workaround for this?

File: src/selectize.js (hunk 1 above)
Comment (r2my) [reply src/selectize.js:1345]: @brookjordan I confirm, your code works fine.

**Comment by colalex** [comment]
Is this still open? 
I have the problem only on Android (other system work fine even with chrome) when I want to delete a selected item.

**Comment by andrey-lizunov** [comment]
@colalex This issue is not about deleting a selected item.
It's about clearing (generate new list) options when you already have selected options and your list of options are dynamic (ajax from server).

**Comment by ravins** [comment]
still open ?

**Comment by kobykorman** [comment]
@joallard Any chance you can merge this? A couple of these really small fixes for impactful UX issues are hanging around, and we could really use your help.

**Comment by joallard** [comment]
Had to rebase that one

**Comment by kstratis** [comment]
2021 and this is still an issue... unbelievable...

**Comment by risadams** [comment]
> 
> 
> 2021 and this is still an issue... unbelievable...

Would you believe that a PR closed two years ago might not be quite relevant to today's environment?  Feel free to open a new issue with a working example, or submit your own PR with the changes you wish to see.

**Pull request closed without merging**